### PR TITLE
Initial version.

### PR DESCRIPTION
**Pull request by davidlehn**
There's an open question about what decode should return:
- Always Uint8Array: simpler code and docs
- Buffer if available, else Uint8Array: docs more complicated, tests need to check this works, opens question if this package should disable Buffer polyfill so Uint8Array would be used, or if dependants decide if they want to use Buffer polyfill in browsers/etc.

### CODE REVIEW

File: baseN.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      carry >>= 8;
+    }
+  }
+
+  // deal with leading zeros
+  for(let k = 0; input[k] === first && k < input.length - 1; ++k) {
+    bytes.push(0);
+  }
+
+  // FIXME: remove
```
Comment (mattcollier) [inline baseN.js]: remove?

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2011-2019 Digital Bazaar, Inc. All rights reserved.
```
Comment (mattcollier) [inline karma.conf.js]: bump date.

File: karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    ],
+
+    // list of files to exclude
+    exclude: [],
+
+    // preprocess matching files before serving them to the browser
+    // available preprocessors:
+    // https://npmjs.org/browse/keyword/karma-preprocessor
+    preprocessors: {
+      //'tests/*.js': ['webpack', 'babel'] //preprocessors
```
Comment (mattcollier) [inline karma.conf.js]: remove?

Comment (mattcollier) [reply karma.conf.js]: general clean-up in this file?

File: main.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
```
Comment (mattcollier) [inline main.js]: bump date.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "base58-universal",
+  "version": "1.0.0-0",
```
Comment (mattcollier) [inline package.json]: this needs to be 0.0.1 in advance of release.

File: baseN.js (hunk 1 above)
Comment (davidlehn) [reply baseN.js]: I did in fact mean to say "remove?" there.  It's related to the question in the PR about using Buffer or not.

Comment (mattcollier) [reply baseN.js]: Sounds like we're rolling with the current implementation.

Comment (davidlehn) [reply baseN.js]: Removed the commented code.  Just going with Uint8Array support for now.

**Pull request merged**